### ISSUE TITLE
AnimationTree suddenly became unusable

### ISSUE THREAD

**Issue by slapin** [godotengine/godot#24933]
After changes to AnimationTree in a set of updates (the transition to properties) it became impossible to use as property values are impossible to distinguish. It is possible to do on top level only, as there with some guesswork one can find proper value $AnimationTree["parameters/node/val"] = ...
However when adding level like state machine or blendtree all ways to guess proper prop name fail. The only way to find them is recursively dump all tree props or just add debug line in C++ code which is not nice at all. Documentation at https://docs.godotengine.org/en/latest/tutorials/animation/animation_tree.html is incorrect or misleading, as hovering over props only report the prop name, not complete path, so even this wacky way of finding parameters is not available.

In addition, ways to activate nodes are absent in any documentation and have to be distinguished from C++ source code and not possible to guess at all. I ended up having to revert all the patches which change to props use, as it takes days to set up AnimationTree and making it work. I consider it serious desing flaw.
You should not have to determine usage of something like this from source code only.

**Comment by akien-mga** [comment]
Did you mean to close it?

**Comment by slapin** [comment]
Yes, as it was too emotional. And was not actually correct, as I
caught 2 different bugs there - node rename problem and property
visibility problem. First one is a bug, second one is quality of life
problem and might not be up for beta stage anyway. It is my problem it
was overwhelming as I had less understanding of animation tree inner
working than now and had trouble looking things up in source code. Now
as I remember all the handles I need to tweak it does not feel like
big deal and feels more or less controllable. Sorry about that.

I guess usability here might be concern for 3.2 or later though. For
now, quickly looking at text resource is enough to see all the tree
structure with all set props. All other props should be looked at C++
source (mostly ones like OneShot activate, etc.). If rename bug does
not hit you, some of these are available in "AnimationTree" node
inspector, but not all for some reason. So there is no single place to
look-up all properties and activation controls except source code.
This might be show stopper for some people.
